### DISCUSSION TITLE
How to set the indentation to 4 spaces?

### DISCUSSION THREAD

**Discussion by FriendsZonn** [LazyVim/LazyVim#2378] [General]
How can I set the tab indentation of lazyvim to 4 spaces? I added the following code in options.lua:

local opt = vim.opt
opt.shiftwidth = 4

When I am writing code, the tab indentation appears to be 4 spaces, but when I save the file with :w, the indentation automatically changes to 2 spaces. Additionally, when I use the command nvim ./file.c to open the file.c file, the tab indentation in that file also changes to 2 spaces. I want the tab indentation for all files to be consistently 4 spaces. What should I do?

**Comment by dpetka2001** [comment]
The option you set is the correct way to do it. When you save the file, the formatter kicks in and formats the file according to the formatter's configuration. For lua files, this is defined in `stylua.toml` and if you open that file you will see that it has an option `indent_width = 2`. This is the reason it formats the file back to 2 spaces instead of what you've already set in your `options.lua` file.

You should either change that option in the configuration file for the corresponding formatter that you're using or you could disable formatting on save with `<leader>uf`, which toggles it on/off but only for the current session of Neovim. If you would like the change to be permanent, you can set `vim.g.autoformat = false` in your `options.lua` file and only call format manually with `<leader>cf`, but the latter will still respect the configuration of the formatter itself, so you should take that into consideration and change also the formatter's settings accordingly.

**Reply by FriendsZonn** [reply]
I entered <leader>cm, and in the Formatter section, it showed that I hadn't installed anything. I added vim.g.autoformat in options.lua, and now the issue no longer occurs. Thank you for resolving my problem.
